File: qif4j/src/main/java/com/github/squirrelgrip/qif4j/write/QifRecord.java
package com.github.squirrelgrip.qif4j.write;

public interface QifRecord {
	/**
	 * Output record in QIF format. If the previous header matches this header
	 * then printing of this header will be suppressed.
	 * 
	 * @param previousHeader
	 *            the header of the previous QifRecord in the stream. May be
	 *            null
	 * @return formatted record
	 */
	String asFormattedRecord(QifHeaderEnum previousHeader);

	/**
	 * Return the header of the QIF Record. For example the record describing
	 * the purchase or sale of a security will have the header:
	 * <code>!Type:Invst</code>
	 * 
	 * @return header
	 */
	QifHeaderEnum getHeader();

}

File: qif4k/src/main/kotlin/com/github/squirrelgrip/qif/write/QifRecord.kt
package com.github.squirrelgrip.qif.write

interface QifRecord {
    /**
     * Output record in QIF format. If the previous header matches this header
     * then printing of this header will be suppressed.
     *
     * @param previousHeader
     * the header of the previous QifRecord in the stream. May be
     * null
     * @return formatted record
     */
    fun asFormattedRecord(previousHeader: QifHeaderEnum?): String

    /**
     * Return the header of the QIF Record. For example the record describing
     * the purchase or sale of a security will have the header:
     * `!Type:Invst`
     *
     * @return header
     */
    val header: QifHeaderEnum?
}